include/h2o.h

```diff
@@ -1563,6 +1563,11 @@ typedef struct st_h2o_compress_context_t {
      */
     void (*compress)(struct st_h2o_compress_context_t *self, h2o_iovec_t *inbufs, size_t inbufcnt, h2o_send_state_t state,
                      h2o_iovec_t **outbufs, size_t *outbufcnt);
+    /**
+     * decompress
+     */
+    void (*decompress)(struct st_h2o_compress_context_t *self, h2o_iovec_t *inbufs, size_t inbufcnt, h2o_send_state_t state,
+                     h2o_iovec_t **outbufs, size_t *outbufcnt);
```

Comment:
I preffer your first option too. The name may be misleading, but I believe it's most reasonable for this issue.